### DISCUSSION TITLE
Add limit param to freeform/purge/spam

### DISCUSSION THREAD

**Discussion by bartdigitalpulse** [solspace/craft-freeform#1020] [Feature Requests]
Upgrading an older project and it seems to have about  > 20.000 spam submissions. When I enable the automatic cleanup for the spam the queue gets stuck. Even when I run "freeform/purge/spam" from the console it takes really long. Any way to makes this query more efficient? If not perhaps add a limit option? Then it can be purged in batches.

**Comment by kjmartens** [comment]
Thank you for your feedback, @bartdigitalpulse. We'll look into improvements for this. 🙂
